smartsim/settings/base.py

```diff
@@ -533,25 +526,24 @@ def set(
 
     @staticmethod
     def _build_exe_args(exe_args: t.Optional[t.Union[str, t.List[str]]]) -> t.List[str]:
-        """Convert exe_args input to a desired collection format"""
+        """Check and convert exe_args input to a desired collection format"""
+        if isinstance(exe_args, list):
+            exe_args = copy.deepcopy(exe_args)
+
+        if not (
```

Comment:
Thanks for the refactor.  I think this is much easier to read.  One minor critique is that I might just have at the beginning of the method:

```python
if not exe_args:
    return []
```

And that at the end of the function you can just return exe_args


Otherwise this is really good

Reply:
Might as well not go through the checks if empty!